### DISCUSSION TITLE
Query

### DISCUSSION THREAD

**Discussion by Dcas89** [zhudotexe/kani#25] [General]
Hi,

How would you add support for azure openai?
How to approach multi-agent conversations?

Many thanks

**Comment by zhudotexe** [comment]
## Azure OpenAI
You can specify a custom endpoint and additional client args when constructing an OpenAIEngine, like so:
```python
engine = OpenAIEngine(
  model="...",
  # override the client's API base
  api_key=azure_api_key,
  api_base=azure_api_base,
  # send extra info in each JSON payload to OpenAI
  engine=azure_deployment_name,
)
```

If needed, you can also pass a `headers={...}` dict into this constructor as well to specify the request headers.

## Multi-agent Conversations
I don't think there's any well-established convention for this yet, but you might find the sub-kani documentation (https://kani.readthedocs.io/en/latest/advanced/subkani.html) helpful - the paradigm here being that messages to another "agent" take  the form of a function call (which you can then process and route however you need).
